src/isomorphic/React.js

```diff
@@ -86,4 +86,31 @@ var React = {
 
 };
 
+if (__DEV__) {
+  // React.DOM factories are deprecated. Set up a Proxy to listen for
+  // accesses to the React.DOM namespace and alert users to switch to
+  // the `react-addons-dom-factories` package.
+  if (typeof Proxy === 'function') {
+    var hasWarnedOfFactories = false;
```

Comment:
Using `Object.defineProperty` could be handy.
With `Object.defineProperty` it might look like this:
```
if (__DEV__) {
  // React.DOM factories are deprecated. ...(rest of comment)
  var hasWarnedOfFactories = false;
  var DOMFactoryMethodNameList = ['a', 'abbr', 'address', ... 'text', 'tspan'];
  DOMFactoryMethodNameList.forEach((method) => {
    React.DOM = Object.defineProperty(React.DOM, method, {
      get: function() {
        if (!hasWarnedOfFactories) {
          warning(
            false,
            'Accessing factories like React.DOM.%s has been deprecated ' +
            'and will be removed in the future. Use the ' +
            'react-addons-dom-factories package instead.',
            method
          );

          hasWarnedOfFactories = true;
        }
        return React.DOM[method];
      },
    });
  });

  }
```
Unless there is a way to define a generic 'get' for all object attributes with `Object.defineProperty`?

If not then it seems just as easy to do without the `Object.defineProperty` call I think, something like this;
```
if (__DEV__) {
  // React.DOM factories are deprecated. ...(rest of comment)
  var hasWarnedOfFactories = false;
  var DOMFactoryMethodNameList = ['a', 'abbr', 'address', ... 'text', 'tspan'];
  DOMFactoryMethodNameList.forEach((method) => {
    var wrappedMethod = React.DOM[method];
    React.DOM[method] = function() {
      if (!hasWarnedOfFactories) {
        warning(
          false,
          'Accessing factories like React.DOM.%s has been deprecated ' +
          'and will be removed in the future. Use the ' +
          'react-addons-dom-factories package instead.',
          method
        );

        hasWarnedOfFactories = true;
      }
      return wrappedMethod.apply(null, Array.prototype.slice.call(arguments));
    });
  });

  }
```
Both of these instantiate an extra wrapper function for every factory method, which is annoying. I figure a follow-up PR, either by me or someone else, can come up with a good approach.